waiter/integration/waiter/basic_test.clj

```diff
@@ -379,6 +392,71 @@
     ;; PORT2 is running kitchen without SSL enabled
     (run-backend-proto-service-test waiter-url "h2" "https" 1 "https" "HTTP/2.0")))
 
+(defn- run-trailers-support-test
+  [waiter-url backend-proto]
+  (testing "request and response trailers"
+    (let [request-length 100000
+          long-request (apply str (repeat request-length "a"))
+          sediment-command (sediment-server-command "${PORT0}")
+          request-headers {:x-waiter-backend-proto backend-proto
+                           :x-waiter-cmd sediment-command
+                           :x-waiter-mem 512
+                           :x-waiter-name (rand-name)}
+          {:keys [service-id] :as canary-response}
+          (make-request-with-debug-info request-headers #(make-shell-request waiter-url % :path "/trailers"))]
+      (with-service-cleanup
+        service-id
+        (is service-id)
+        (assert-response-status canary-response 200)
+
+        (doseq [response-status [200 400 500]]
+          (doseq [request-trailer-delay-ms [0 5000]]
+            (doseq [response-trailer-delay-ms [0 5000]]
+
+              (testing (str {:backend-proto backend-proto
+                             :request-trailer-delay-ms request-trailer-delay-ms
+                             :response-status response-status
+                             :response-trailer-delay-ms response-trailer-delay-ms})
+
+                (let [request-trailers {(rand-name "foo") (rand-name "bar")
+                                        (rand-name "lorem") (rand-name "ipsum")}
+                      response-trailers {(rand-name "fee") (rand-name "fie")
+                                         (rand-name "foe") (rand-name "fum")}
+                      response (make-shell-request
+                                 waiter-url
+                                 (reduce
+                                   (fn [request-headers [k v]]
+                                     (assoc request-headers
+                                       (str "x-sediment-response-trailer-" k) (str v)))
+                                   (assoc request-headers
+                                     "x-sediment-response-status" response-status
+                                     "x-sediment-sleep-before-response-trailer-ms" response-trailer-delay-ms)
+                                   (seq response-trailers))
+                                 :body (ByteArrayInputStream. (.getBytes long-request))
+                                 :path "/trailers"
+                                 :protocol backend-proto
+                                 :trailers-fn (fn []
+                                                (Thread/sleep request-trailer-delay-ms)
```

Comment:
I would love to confirm through other means that this sleep introduces the delay in the correct part of the stream.

Reply:
I haven't confirmed this yet, will look into it and let you know what I find.